packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
@mjackson from twitter it appears that you aren't the biggest fan of HOCs, but could a `listen` HOC be useful for something like that? I believe that component composition could work for this to.

I.e., You can remove almost all of the listeners and by default only the `<Router>` listens for changes and triggers updates. For most users everything will propagate correctly However, if you have a component that implements `shouldComponentUpdate`, you can use a `listen` HOC that will listen for and re-render that component on location changes.

```js
import { listen, Listen } from 'react-router-listen'

const MyComponent = listen((props) => (
    <Link to='some/place'>{props.superPerformant}</Link>
))

const MySpecialComponent  = connect(
  () => ({ superPerformant: true })
)(MyComponent)

// or

class MySpecialComponent extends React.Component {
  shouldComponentUpdate() { return false }

  render() {
    return (
      <Listen>
        {() => <Link to='some/place'>Some Place</Link>}
      </Listen>
  )
}
```
Maybe that is asking too much, but I think that it could work.
:man_shrugging: